symengine/ruby/ext/symengine/ruby_basic.c

```diff
@@ -5,23 +5,25 @@ void cbasic_free(void *ptr){
     basic_free(basic_ptr);
 }
 
-VALUE cbasic_alloc(VALUE klass){
+void cbasic_free_heap(void *ptr) {
+    basic_struct *basic_ptr = ptr;
+    basic_free_heap(basic_ptr);
+}
+
+VALUE alloc_func(VALUE klass, void(*free_func_ptr)(void *)) {
```

Comment:
We have only one method for allocation right? So, this can just be `cbasic_alloc`